### PR TITLE
fix log monitor read error

### LINKED ISSUE #5220: log monitor report UnicodeDecodeError
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: centos 7
- **Ray installed from (source or binary)**: binay
- **Ray version**: 0.7.2 0.8
- **Python version**: python 3.7
- **Exact command to reproduce**:

### Describe the problem

Now the `log_monitor` monitor all the files under logs. This could be causing the following errors when we read those log file with `VIM` because of the `.*.swap` file will be created. 
```Traceback (most recent call last):
  File "/home/xianyang/anaconda3/lib/python3.7/site-packages/ray/log_monitor.py", line 278, in <module>
    raise e
  File "/home/xianyang/anaconda3/lib/python3.7/site-packages/ray/log_monitor.py", line 268, in <module>
    log_monitor.run()
  File "/home/xianyang/anaconda3/lib/python3.7/site-packages/ray/log_monitor.py", line 219, in run
    anything_published = self.check_log_files_and_publish_updates()
  File "/home/xianyang/anaconda3/lib/python3.7/site-packages/ray/log_monitor.py", line 175, in check_log_files_and_publish_updates
    next_line = file_info.file_handle.readline()
  File "/home/xianyang/anaconda3/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xca in position 21: invalid continuation byte

```

### PR DESCRIPTION
**Pull request by ConeyLiu**
## What do these changes do?
Currently, we will encounter some errors when monitoring all the file under logs. And we also just need the worker logfile from the code. In this patch, we only read those file which matches a worker log file format.

## Related issue number

Closes #5220

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by ConeyLiu** [comment]
`scripts/format.sh ` failed as: ```flake8: error: no such option: --inline-quotes```

**Comment by richardliaw** [comment]
`pip install flake8-quotes`

File: python/ray/log_monitor.py [hunk 1]
```diff
@@ -93,7 +101,9 @@ def update_log_filenames(self):
 
         for log_filename in log_filenames:
             full_path = os.path.join(self.logs_dir, log_filename)
-            if full_path not in self.log_filenames:
+            filename = full_path.split("/")[-1]
```
Comment (richardliaw) [inline python/ray/log_monitor.py]: os.path.basename?

File: python/ray/log_monitor.py [hunk 2]
```diff
@@ -93,7 +101,9 @@ def update_log_filenames(self):
...
+            if (file_should_monitor(filename)
```
Comment (richardliaw) [inline python/ray/log_monitor.py]: this won't even run right? `self.file_should_monitor`

Comment (richardliaw) [reply python/ray/log_monitor.py]: can you use glob instead?

**Comment by ConeyLiu** [comment]
thanks @richardliaw for the comments. Is there any way to running the UT tests in locally? I have run the 
`pip install -e . --verbose` and got the following errors. It seems not related.
```
ERROR: /home/lxy/git_repository/ray/BUILD.bazel:718:1: Executing genrule //:redis failed (Exit 2) bash failed: error executing command
      (cd /home/lxy/.cache/bazel/_bazel_lxy/92fa78d6e2702ccf396ac1998b6e1ae8/sandbox/linux-sandbox/4845/execroot/com_github_ray_project_ray && \
      exec env - \
        LD_LIBRARY_PATH=/usr/local/lib/:/usr/local/lib/:/usr/local/lib/: \
        PATH=/home/lxy/utils/gradle-4.0/bin:/home/lxy/anaconda3/bin:/home/lxy/utils/scala-2.11.8/bin:/home/lxy/utils/maven/bin:/home/lxy/utils/jdk1.8.0_111/bin:/home/lxy/bin:/home/lxy/.local/bin:/home/lxy/utils/gradle-4.0/bin:/home/lxy/anaconda3/bin:/home/lxy/utils/scala-2.11.8/bin:/home/lxy/utils/maven/bin:/home/lxy/utils/jdk1.8.0_111/bin:/home/lxy/bin:/home/lxy/.local/bin:/home/lxy/utils/gradle-4.0/bin:/home/lxy/anaconda3/bin:/home/lxy/utils/scala-2.11.8/bin:/home/lxy/utils/maven/bin:/home/lxy/utils/jdk1.8.0_111/bin:/home/lxy/bin:/home/lxy/.local/bin:/home/lxy/utils/gradle-4.0/bin:/home/lxy/anaconda3/bin:/home/lxy/utils/scala-2.11.8/bin:/home/lxy/utils/maven/bin:/home/lxy/utils/jdk1.8.0_111/bin:/home/lxy/bin:/home/lxy/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/lxy/utils/sbt/bin:/home/lxy/utils/sbt/bin:/home/lxy/utils/sbt/bin:/home/lxy/bin:/home/lxy/utils/sbt/bin:/home/lxy/bin \
        PYTHON_BIN_PATH=/home/lxy/anaconda3/bin/python3 \
      /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh;
            set -x &&
            curl -sL "https://github.com/antirez/redis/archive/5.0.3.tar.gz" | tar xz --strip-components=1 -C . &&
            make &&
            mv ./src/redis-server bazel-out/k8-opt/bin/redis-server &&
            chmod +x bazel-out/k8-opt/bin/redis-server &&
            mv ./src/redis-cli bazel-out/k8-opt/bin/redis-cli &&
            chmod +x bazel-out/k8-opt/bin/redis-cli
        ')
    Execution platform: @bazel_tools//platforms:host_platform
```

File: python/ray/log_monitor.py [hunk 3]
```diff
@@ -89,18 +90,20 @@ def close_all_files(self):
...
     def update_log_filenames(self):
         """Update the list of log files to monitor."""
-        log_filenames = os.listdir(self.logs_dir)
-
-        for log_filename in log_filenames:
-            full_path = os.path.join(self.logs_dir, log_filename)
-            if full_path not in self.log_filenames:
-                self.log_filenames.add(full_path)
+        # we only monior worker log files
+        log_file_paths = glob.glob("{}/worker*[.out|.err]".format(
```
Comment (richardliaw) [inline python/ray/log_monitor.py]: isn't there other log files that the log_monitor tracks? like the monitor log, the raylet log?

File: python/ray/log_monitor.py [hunk 4]
```diff
@@ -195,7 +199,7 @@ def check_log_files_and_publish_updates(self):
             # Record the current position in the file.
             file_info.file_position = file_info.file_handle.tell()
 
-            if len(lines_to_publish) > 0 and is_worker:
+            if len(lines_to_publish) > 0:
                 self.redis_client.publish(
```
Comment (ConeyLiu) [inline python/ray/log_monitor.py]: @richardliaw we only publish worker log file, not sure why. Please correct me if I am not right.

**Comment by robertnishihara** [comment]
@ConeyLiu, thanks for submitting the PR. You're right that we currently only stream the worker logs back to the driver. However, in the future I think it would make sense to collect all of the process logs and store them in some central database or something like that.

The PR looks good to me.

**Review by robertnishihara — APPROVED** [review]

**Comment by ConeyLiu** [comment]
thanks a lot @richardliaw @robertnishihara.

**Pull request merged**